### PR TITLE
Input parser

### PR DESCRIPTION
**Pull request by yaoandrew**

### CODE REVIEW

File: lib/hand.rb [hunk 1]
```diff
@@ -1,6 +1,16 @@
-require 'deck'
+require_relative 'deck'
```
Comment (catmclough) [inline lib/hand.rb]: Think we can get rid of this dependency 😃

File: lib/hand.rb [hunk 2]
```diff
@@ -10,77 +20,59 @@ def add_card(card)
...
-    card_occurences.select { |_k, v| v == 2}.count == 2
-  end
-
-  def contains_straight?
-    card_sum == consecutive_sum || card_sum == 28 && all_unique?
-  end
-
-  def contains_flush?
-    suit_occurences.value?(5)
+  def contains_card?(testcard)
```
Comment (catmclough) [inline lib/hand.rb]: I think the name `testcard` is a bit confusing. How about just using `card` here? You'd have to use something else for the card in the hand with which you're comparing it to on line 28, though. Maybe call that one something like `existing_card` or `card_in_hand`?

Comment (yaoandrew) [reply lib/hand.rb]: Yeah.. that definitely makes it a bit more readable.

File: lib/hand.rb (hunk 1 above)
Comment (yaoandrew) [reply lib/hand.rb]: Ha ha.. gone.

**Pull request merged**